scripts/genertate/calculate_metrics.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+import argparse
+import sacrebleu
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='')
+    parser.add_argument('--generated_file', type=str, required=True, help='Model name')
+    return parser.parse_args()
+
+
+def calculate_self_bleu4(samples):
```

Comment:
You can use the functions implemented in sacrebleu: https://github.com/mjpost/sacrebleu